centml/compiler/prediction/profiler.py

```diff
@@ -0,0 +1,226 @@
+        }
+
+        def load_arg(a):
+            return torch.fx.graph.map_arg(a, lambda n: env[n.name])
+
+        def fetch_attr(target: str):
+            target_atoms = target.split('.')
+            attr_itr = self.mod
+            for i, atom in enumerate(target_atoms):
+                if not hasattr(attr_itr, atom):
+                    raise RuntimeError(f"Node referenced nonexistant target {'.'.join(target_atoms[:i])}")
+                attr_itr = getattr(attr_itr, atom)
+            return attr_itr
+
+        def get_flattened_shapes(args):
+            flattened_shapes = []
+            dtypes = []
+
+            for arg in args:
+                if isinstance(arg, (tuple, list)):
+                    if len(arg) > 0 and isinstance(arg[0], (tuple, list, torch.Tensor)):
+                        nested_shapes, nested_dtypes = get_flattened_shapes(arg[0])
+                        shape = [len(arg)] + nested_shapes
+                        dtypes.extend(nested_dtypes.split(','))
+                    else:
+                        shape = [len(arg)]
+                elif isinstance(arg, torch.Tensor):
+                    shape = list(arg.shape)
+                    dtypes.append(dtypeMap[arg.dtype])
+                elif isinstance(arg, bool):
+                    shape = [1 if arg is True else 0]
+                elif isinstance(arg, (int, float)):
+                    shape = [arg]
+                else:
+                    shape = [1]
+                flattened_shapes.extend(shape)
+
+            if len(flattened_shapes) < 2:
+                flattened_shapes.extend([1])
+
+            input_dtypes = ','.join(dtypes) if dtypes else 'N/A'
+
+            return flattened_shapes, input_dtypes
+
+        def get_output_dtypes(results):
+            def find_dtypes(results):
+                if isinstance(results, torch.Tensor):
+                    return [dtypeMap[results.dtype]]
+                elif isinstance(results, (list, tuple)):
+                    dtypes = []
+                    for item in results:
+                        dtypes.extend(find_dtypes(item))
+                    return dtypes
+                return []
+
+            types = find_dtypes(results)
+
+            if types:
+                return ','.join(types)
+            return 'N/A'
+
+        for node in self.graph.nodes:
+            result = None
+            if node.op == 'placeholder':
+                result = next(args_iter)
+            elif node.op == 'get_attr':
+                result = fetch_attr(node.target)
+            elif node.op == 'call_function':
+                args = load_arg(node.args)
+                kwargs = load_arg(node.kwargs)
+                result = node.target(*args, **kwargs)
+
+                inp_shapes, input_dtypes = get_flattened_shapes(args)
+                output_dtypes = get_output_dtypes(result)
+
+                key = (node.target.__name__, len(inp_shapes), input_dtypes, output_dtypes, self.gpu)
```

Comment:
```suggestion
                key = (node.target, len(inp_shapes), input_dtypes, output_dtypes, self.gpu)
```

Reply:
To be consistent with the rest.

Reply:
It's not a choice, sometimes node.target is a string, sometimes it's an object. In the case of a call_function it's an object. I can try cast them both to a str see if that works